chalk-solve/src/clauses.rs

```diff
@@ -262,31 +266,35 @@ fn program_clauses_that_could_match<I: Interner>(
             match_ty(builder, environment, ty)
         }
         DomainGoal::FromEnv(_) => (), // Computed in the environment
-        DomainGoal::Normalize(Normalize { alias, ty: _ }) => {
-            // Normalize goals derive from `AssociatedTyValue` datums,
-            // which are found in impls. That is, if we are
-            // normalizing (e.g.) `<T as Iterator>::Item>`, then
-            // search for impls of iterator and, within those impls,
-            // for associated type values:
-            //
-            // ```ignore
-            // impl Iterator for Foo {
-            //     type Item = Bar; // <-- associated type value
-            // }
-            // ```
-            let associated_ty_datum = db.associated_ty_data(alias.associated_ty_id);
-            let trait_id = associated_ty_datum.trait_id;
-            let trait_parameters = db.trait_parameters_from_projection(alias);
-            push_program_clauses_for_associated_type_values_in_impls_of(
-                builder,
-                trait_id,
-                trait_parameters,
-            );
-        }
+        DomainGoal::Normalize(Normalize { alias, ty: _ }) => match alias {
+            AliasTy::Projection(proj) => {
+                // Normalize goals derive from `AssociatedTyValue` datums,
+                // which are found in impls. That is, if we are
+                // normalizing (e.g.) `<T as Iterator>::Item>`, then
+                // search for impls of iterator and, within those impls,
+                // for associated type values:
+                //
+                // ```ignore
+                // impl Iterator for Foo {
+                //     type Item = Bar; // <-- associated type value
+                // }
+                // ```
+                let associated_ty_datum = db.associated_ty_data(proj.associated_ty_id);
+                let trait_id = associated_ty_datum.trait_id;
+                let trait_parameters = db.trait_parameters_from_projection(proj);
+                push_program_clauses_for_associated_type_values_in_impls_of(
+                    builder,
+                    trait_id,
+                    trait_parameters,
+                );
+            }
+            AliasTy::Opaque(_) => (),
```

Comment:
It seems like we're missing something here. I would expect if we have the goal

```rust
Normalize(OpaqueTy -> ...)
```

that we would generate clauses from the opaque type definition to the hidden type (but only if `Reveal` is in the environment).

Reply:
(Never mind, I figured this out later)